src/net.cpp

```diff
@@ -2865,3 +2864,23 @@ uint64_t CConnman::CalculateKeyedNetGroup(const CAddress& ad) const
 
     return GetDeterministicRandomizer(RANDOMIZER_ID_NETGROUP).Write(vchNetGroup.data(), vchNetGroup.size()).Finalize();
 }
+
+void LogMessage(const CNode& node, const std::string& msg_type, const Span<const unsigned char>& data, bool is_incoming)
+{
+    int64_t time = GetTimeMicros();
+
+    fs::path base_path = GetDataDir() / "peer_logging" / node.addr.ToString();
+    fs::create_directories(base_path);
+
+    fs::path path = base_path / (is_incoming ? "msgs_recv.dat" : "msgs_sent.dat");
+    CAutoFile f(fsbridge::fopen(path, "ab"), SER_DISK, CLIENT_VERSION);
+
+    f.write((const char*)&time, sizeof(time));
+    f.write(msg_type.data(), msg_type.length());
+    for (int i = msg_type.length(); i < CMessageHeader::COMMAND_SIZE; ++i) {
```

Comment:
```suggestion
    for (auto i = msg_type.length(); i < CMessageHeader::COMMAND_SIZE; ++i) {
```

to fix the compile failure:

```
net.cpp:2880:39: error: comparison of integers of different signs: 'int' and 'const size_t' (aka 'const unsigned long') [-Werror,-Wsign-compare]

    for (int i = msg_type.length(); i < CMessageHeader::COMMAND_SIZE; ++i) {

                                    ~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

1 error generated.

Reply:
I'm not sure why this didn't come up before, and still doesn't come up on my machine :/  Regardless, fixed.

Reply:
I don't know either: #19123